packages/authentication/src/providers/auth-metadata.ts

```diff
@@ -18,9 +18,9 @@ import {
 export class AuthMetadataProvider
   implements Provider<AuthenticationMetadata | undefined> {
   constructor(
-    @inject(CoreBindings.CONTROLLER_CLASS)
+    @inject(RestBindings.CONTROLLER_CLASS)
```

Comment:
Ah, so this is an interesting problem. @bajtos brought up the idea of namespaces (which we could map to the Application and Server concept) that would let us differentiate between a REST server's `"controllers.*"` and a gRPC server's `"controllers.*"` bindings.

Alternatively, we could just make those keys unique so that there's no collision and make it clear to component devs that making their own Servers would require care in this regard.

Reply:
Since the current controller is bound in the per-request context, I think it's ok to have one binding key shared by multiple transport components (gRPC, REST) - I mean it should technically work.

As I mentioned on Slack, in my experience so far, having `app.controller` API provided by the core is feasible, in which case it makes more sense to me to keep `CONTROLLER_CLASS` binding in `CoreBindings` and share it by all transports.

Reply:
Agreed, moving the controller bindings back to core.